src/rgw/services/svc_bi_rados.cc

```diff
@@ -178,8 +180,12 @@ int RGWSI_BucketIndex_RADOS::open_bucket_index(const RGWBucketInfo& bucket_info,
     return ret;
   }
 
+  //auto gen = bucket_info.layout.current_index.gen;
+
+ // TODO: need reshard changes to add gen_id here
```

Comment:
i'm not sure what the gap is here, since you have the whole bucket_index_layout_generation

is there a dependency on your reshard pr?

Reply:
yeah the reshard pr has the index generation passed around in methods like get_bucket_index_objects() for adding it as part of bucket objs. Will need to cherry pick them.

Reply:
have you had a chance to review https://github.com/ceph/ceph/pull/38657? i have a version of that rebased on top of the wip-rgw-multisite-reshard working branch at https://github.com/cbodley/ceph/commits/wip-rgw-multisite-reshard-instance. if you don't see any issues with that PR, i could pull that into the wip-rgw-multisite-reshard branch and you could rebase this one?

Reply:
Sorry I haven't been able to get to it. Will do as soon as I can.